src/mds/Server.cc

```diff
@@ -7273,6 +7367,15 @@ void Server::handle_client_getvxattr(const MDRequestRef& mdr)
       // since we only handle ceph vxattrs here
       r = -ENODATA; // no such attribute
     }
+  } else if (xattr_name == "ceph.dir.subvolume.snaps.visible"sv) {
+    if (!cur->is_dir()) {
+      r = -ENOTDIR;
+    } else if (const auto srnode = cur->get_projected_srnode();
+               !(srnode && srnode->is_subvolume())) {
+      r = -EPERM;
```

Comment:
this needs to be changed to `EINVAL` for uniformity with similar checks in `handle_client_setvxattr`

Reply:
done